### PR TITLE
Fix the path to jacoco report for sonarcloud

### PR DESCRIPTION
**Pull request by ziyadbagh**
#### 1. **Moved the SonarQube Configuration**
Previously, the **SonarQube configuration** block was located at the end of the `build.gradle.kts` file, outside the main `android` configuration. I moved the **SonarQube configuration** into the correct location within the `android` block to ensure that it is properly integrated and applied during the build process. This change aligns the SonarQube setup with the rest of the project configurations.

#### 2. **Updated the Jacoco XML Report Path**
The **`sonar.coverage.jacoco.xmlReportPaths`** property in the SonarQube configuration was updated. Initially, it did not include the **`jacocoTestReport`** folder in the path, which resulted in incorrect report generation.
This ensures that the correct Jacoco test report is generated and included in the SonarQube analysis.

#### 3. **Work on SonarCloud Analysis**
I also did additional work on configuring **SonarCloud** integration, but some methods were not resolving the analysis issues. These adjustments were aimed at improving the connection and results of the analysis with SonarCloud.

#### 4. **Code Cleaning in SonarQube Configuration**
In addition to these changes, I performed some **code cleaning** in the SonarQube configuration by organizing properties such as `sonar.projectKey`, `sonar.organization`, and `sonar.host.url`. This cleanup ensures the configuration is clear and easier to maintain.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -144,6 +144,17 @@ android {
...
+        properties {
+            property ("sonar.projectKey", "Signify-epfl_signify-app")
+            property ("sonar.organization", "signify-epfl")
+            property ("sonar.host.url", "https://sonarcloud.io")
+            property ("sonar.coverage.jacoco.xmlReportPaths", "${project.layout.buildDirectory.get()}/reports/jacoco/jacocoTestReport/jacocoTestReport.xml")
+            property ("sonar.junit.ReportPaths", "${project.layout.buildDirectory.get()}/test-results/testDebugUnitTest/")
+            property ("sonar.androidLint.reportPaths", "${project.layout.buildDirectory.get()}/reports/lint-results-debug.xml")
+        }
+    }
+
```
Comment (samymessouak) [inline app/build.gradle.kts:157]: Nice way to link jacoco to sonarcloud

File: app/build.gradle.kts [hunk 2]
```diff
@@ -144,6 +144,17 @@ android {
         resources.setSrcDirs(emptyList<File>())
     }
```
Comment (SaFouNaldo) [inline app/build.gradle.kts:146]: Add sonar in the build.gradle with correct propreties

**Review by samymessouak — APPROVED** [review]
**Nice work !**
This is a really important step as we can now approve the new code considering the test coverage on new lines.

**Review by SaFouNaldo — APPROVED** [review]
Add sonar in the build.gradle with correct propreties

**Pull request merged**